orderbook/src/main.rs

```diff
@@ -415,21 +415,35 @@ async fn main() {
     };
     check_database_connection(orderbook.as_ref()).await;
 
-    let serve_task = serve_task(
+    let (shutdown_sender, shutdown_receiver) = tokio::sync::oneshot::channel();
+    let serve_api = serve_api(
         database.clone(),
         orderbook.clone(),
         fee_calculator,
         price_estimator,
         args.bind_address,
+        shutdown_receiver,
     );
     let maintenance_task =
         task::spawn(service_maintainer.run_maintenance_on_new_block(current_block_stream));
     let db_metrics_task = task::spawn(database_metrics(metrics, postgres));
 
+    let mut metrics_address = args.bind_address;
+    metrics_address.set_port(DEFAULT_METRICS_PORT);
+    tracing::info!(%metrics_address, "serving metrics");
+    let metrics_task = serve_metrics(orderbook, metrics_address);
+
+    futures::pin_mut!(serve_api);
     tokio::select! {
-        result = serve_task => tracing::error!(?result, "serve task exited"),
+        result = &mut serve_api => tracing::error!(?result, "API task exited"),
         result = maintenance_task => tracing::error!(?result, "maintenance task exited"),
         result = db_metrics_task => tracing::error!(?result, "database metrics task exited"),
+        result = metrics_task => tracing::error!(?result, "metrics task exited"),
+        _ = tokio::signal::ctrl_c() => {
+            tracing::info!("Gracefully shutting down API");
+            shutdown_sender.send(()).expect("failed to send shutdown signal");
+            serve_api.await.expect("API failed during shutdown");
```

Comment:
Might be a good idea to have a timeout here in case for whatever reason a request takes a long time to complete (could be a problem with a client that is out of our control).

Reply:
Kubernetes has a timeout built in to shutting services down 30s after they receive SIGTERM. But sure can add it here as well.

Reply:
Agree we should timeout here so that using `Ctrl-C` on the command line when running locally can't hang for ever.